plan/rule_column_pruning.go

```diff
@@ -132,6 +132,7 @@ func (p *LogicalUnionAll) PruneColumns(parentUsedCols []*expression.Column) {
 	for _, child := range p.Children() {
 		child.PruneColumns(parentUsedCols)
 	}
+	p.schema.Columns = p.children[0].Schema().Columns
```

Comment:
Is this right? parent columns may have different `UniqueID` and `RetType` now?

Reply:
This can be fixed by fixing
>what if all children of LogicalUnionAll have same RetType for each Column? We would still generate new UniqueID for each Column? Can we just use previous UniqueID in this case?

Reply:
Here https://github.com/pingcap/tidb/pull/7680/files#diff-638d8be162d442ef3a5c423ff0c735d4R642 the `UniqueID` of columns in `UnionAll` is different from child[0]

> union's col's id is the same with child's

What does `id` in the above sentence mean?

Reply:
It's done by [here](https://github.com/pingcap/tidb/pull/7680/files#diff-638d8be162d442ef3a5c423ff0c735d4R669)